code/edgeidentity/src/main/java/com/adobe/marketing/mobile/edge/identity/URLUtils.java

```diff
@@ -54,7 +54,7 @@ static String generateURLVariablesPayload(final String ts, final String ecid, fi
 				// No need to encode
 				urlFragment.append("null");
 			} else {
-				urlFragment.append(URLEncoder.encode(theIdString, StandardCharsets.UTF_8.toString()));
+				urlFragment.append(URLEncoder.encode(theIdString, Charset.forName("UTF-8").name()));
```

Comment:
Can you pass a Charset as the second argument instead of a String, using the function `URLEncoder.encode(String s, Charset charset)`? For example `URLEncoder.encode(theIdString, StandardCharsets.UTF_8)`?
https://developer.android.com/reference/java/net/URLEncoder#encode(java.lang.String,%20java.nio.charset.Charset)

Reply:
Ah, sorry, that API was added in Android 33 so we can't use it.

The change looks good, however `Charset.forName()` can throw a number of exceptions. There is a low chance that the device doesn't support `UTF-8`, but to safeguard ourselves against the unknown, we should handle the `IllegalCharsetException` and `UnsupportedCharsetException`.
https://developer.android.com/reference/java/nio/charset/Charset#forName(java.lang.String)